actions/orchestration.go

```diff
@@ -10,7 +10,224 @@ import (
+				}
+
+				return nil, ErrCode("failed to create database instance", err)
+			}
+
+			log.Println("created RDS instance", instanceOutput)
+
+			resp.Instance = instanceOutput.DBInstance
+		}
+	} else if req.Instance != nil {
+		snapshotId = aws.StringValue(req.Instance.SnapshotIdentifier)
+		if snapshotId == "" {
+			return nil, errors.New("empty snapshot identifier")
+		}
+
+		if req.Instance.DBInstanceIdentifier == nil {
+			return nil, errors.New("empty DBInstanceIdentifier")
+		}
+
+		// get information about the snapshot
+		var snapshot *rds.DBSnapshot
+		snapshotsOutput, err := o.client.Service.DescribeDBSnapshotsWithContext(c, &rds.DescribeDBSnapshotsInput{
+			DBSnapshotIdentifier: aws.String(snapshotId),
+		})
+		if err != nil {
+			return nil, err
+		} else if len(snapshotsOutput.DBSnapshots) > 1 {
+			return nil, errors.New("unexpected number of snapshots")
+		} else {
+			snapshot = snapshotsOutput.DBSnapshots[0]
+		}
+
+		req.Instance.Tags = normalizeTags(req.Instance.Tags)
+
+		// set default subnet group
+		if req.Instance.DBSubnetGroupName == nil {
+			req.Instance.DBSubnetGroupName = aws.String(o.client.DefaultSubnetGroup)
+		}
+
+		// set default parameter group
+		if req.Instance.DBParameterGroupName == nil {
+			pgFamily, pgErr := o.client.DetermineParameterGroupFamily(snapshot.Engine, snapshot.EngineVersion)
+			if pgErr != nil {
+				log.Println(pgErr.Error())
+				return nil, pgErr
+			}
+			log.Println("determined ParameterGroupFamily based on Engine:", pgFamily)
+			if pg, ok := o.client.DefaultDBParameterGroupName[pgFamily]; ok {
+				log.Println("using DefaultDBParameterGroupName:", pg)
+				req.Instance.DBParameterGroupName = aws.String(pg)
+			}
+		}
+
+		input := &rds.RestoreDBInstanceFromDBSnapshotInput{
+			AutoMinorVersionUpgrade:     aws.Bool(true),
+			CopyTagsToSnapshot:          aws.Bool(true),
+			DBInstanceIdentifier:        req.Instance.DBInstanceIdentifier,
+			DBParameterGroupName:        req.Instance.DBParameterGroupName,
+			DBSnapshotIdentifier:        aws.String(snapshotId),
+			DBSubnetGroupName:           req.Instance.DBSubnetGroupName,
+			EnableCloudwatchLogsExports: req.Instance.EnableCloudwatchLogsExports,
+			MultiAZ:                     req.Instance.MultiAZ,
+			Port:                        req.Instance.Port,
+			PubliclyAccessible:          aws.Bool(false),
+			Tags:                        toRDSTags(req.Instance.Tags),
+			VpcSecurityGroupIds:         req.Instance.VpcSecurityGroupIds,
+		}
+
+		log.Printf("restoring database instance: %+v", *input)
+
+		output, err := o.client.Service.RestoreDBInstanceFromDBSnapshotWithContext(c, input)
+		if err != nil {
+			return nil, ErrCode("failed to create database instance from snapshot", err)
+		}
+
+		log.Printf("created RDS instance from snapshot: %+v", output.DBInstance)
+
+		resp.Instance = output.DBInstance
+	} else {
+		return nil, errors.New("invalid request")
+	}
```

Comment:
instead of this `else if` and `else`, can you just return inside the `if req.Cluster != nil` and `if req.Instance != nil` blocks?  then this is the fall through at the end of the function.